src/controllers/wallet/atomic-swap/tx-proposal.controller.js

```diff
@@ -52,9 +52,15 @@ async function buildTxProposal(req, res) {
     return;
   }
 
-  const proposal = partialTx
-    ? PartialTxProposal.fromPartialTx(partialTx, wallet.storage)
-    : new PartialTxProposal(wallet.storage);
+  let proposal;
+  try {
+    proposal = partialTx
+      ? PartialTxProposal.fromPartialTx(partialTx, wallet.storage)
+      : new PartialTxProposal(wallet.storage);
+  } catch (e) {
+    res.status(400).json({ success: false, error: 'Invalid serialized partial_tx' });
```

Comment:
Isn't it better to return this only if we are sure the error was caused by an invalid partial_tx?

Maybe checking the exception type to be `SyntaxError` or `UnsupportedScriptError` would achieve this.